src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -823,11 +823,21 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         const localDeviceId = this.client.getDeviceId();
         if (!localUserId || !localDeviceId) throw new Error("User ID or device ID was null!");
 
-        const myCallMemberEvent = roomState.getStateEvents(EventType.GroupCallMemberPrefix, localUserId) ?? undefined;
-        const content = myCallMemberEvent?.getContent() ?? {};
-        const legacy = "memberships" in content || this.useLegacyMemberEvents;
+        const callMemberEvents = roomState.events.get(EventType.GroupCallMemberPrefix);
+        let legacy = !!this.useLegacyMemberEvents;
+        if (!legacy && callMemberEvents?.size) {
+            for (const callMemberEvent of callMemberEvents.values()) {
+                const content = callMemberEvent.getContent();
+                if (Array.isArray(content["memberships"]) && content["memberships"].length > 0) {
+                    legacy = true;
+                    break;
+                }
+            }
+        }
         let newContent: {} | ExperimentalGroupCallRoomMemberState | SessionMembershipData = {};
         if (legacy) {
+            const myCallMemberEvent = callMemberEvents?.get(localUserId);
+            const content = myCallMemberEvent?.getContent() ?? {};
             let myPrevMembership: CallMembership | undefined;
             // We know its CallMembershipDataLegacy
```

Comment:
I think if we already do it properly we also need to check for the expiration of those events.

```suggestion
        const callMemberEvents = roomState.events.get(EventType.GroupCallMemberPrefix);
        const legacy =
            !!this.useLegacyMemberEvents ||
            (callMemberEvents?.size && this.stateEventsContainOngoingLegacySession(callMemberEvents));
        let newContent: {} | ExperimentalGroupCallRoomMemberState | SessionMembershipData = {};
        if (legacy) {
            const myCallMemberEvent = callMemberEvents?.get(localUserId);
            const content = myCallMemberEvent?.getContent() ?? {};
            let myPrevMembership: CallMembership | undefined;
            // We know its CallMembershipDataLegacy
```

and then add:
```
    private stateEventsContainOngoingLegacySession(callMemberEvents: Map<string, MatrixEvent>){
       return Array.from(callMemberEvents.values()).some((memberEvent)=>{
            const content = memberEvent.getContent();
            return Array.isArray(content["memberships"]) &&
            content.memberships.map((m) => new CallMembership(memberEvent, m)).filter((m)=>m.isExpired()).length > 0;
        })
    }
```